slick/src/main/scala/slick/util/AsyncExecutor.scala

```diff
@@ -37,12 +39,14 @@ object AsyncExecutor extends Logging {
           case 0 => new SynchronousQueue[Runnable]
           case -1 => new LinkedBlockingQueue[Runnable]
           case n =>
-            new ManagedArrayBlockingQueue[Runnable](n * 2) {
-              def accept(r: Runnable, size: Int) = r match {
-                case pr: PrioritizedRunnable if pr.highPriority => true
-                case _ => size < n
+            val q = new ManagedArrayBlockingQueue[Runnable](n) {
+              override protected[this] def priority(item: Runnable): Priority = item match {
+                case pr: PrioritizedRunnable => pr.priority
+                case _ => LowPriority
               }
             }
+            pausable = Option(q)
```

Comment:
pausable is initialized as None and only re-assigned once when the executionContext is created.